packages/apps/halo-app/src/main.tsx

```diff
@@ -28,43 +29,54 @@ const HaloMeta = {
   id: 'dxos.org/plugin/halo-app',
 };
 
-const App = createApp({
-  fallback: (
-    <div className='flex h-screen justify-center items-center'>
-      <ProgressBar indeterminate />
-    </div>
-  ),
-  order: [TelemetryMeta, ThemeMeta, PwaMeta, ErrorMeta, ClientMeta, HaloMeta],
-  plugins: {
-    [TelemetryMeta.id]: Plugin.lazy(() => import('@braneframe/plugin-telemetry'), {
-      namespace: appKey,
-      config: new Config(Defaults()),
-    }),
-    [ThemeMeta.id]: Plugin.lazy(() => import('@braneframe/plugin-theme'), { appName: 'HALO' }),
-    // Outside of error boundary so that updates are not blocked by errors.
-    [PwaMeta.id]: Plugin.lazy(() => import('@braneframe/plugin-pwa')),
-    // Inside theme provider so that errors are styled.
-    [ErrorMeta.id]: Plugin.lazy(() => import('@braneframe/plugin-error')),
-    [ClientMeta.id]: Plugin.lazy(() => import('@braneframe/plugin-client'), { appKey }),
-    [HaloMeta.id]: async () =>
-      ({
-        meta: HaloMeta,
-        provides: {
-          translations,
-          root: () => {
-            return (
-              <div className='flex h-screen justify-center items-center'>
-                <OpenVault />
-              </div>
-            );
+const main = async () => {
+  const config = new Config(await Envs(), Local(), Defaults());
+  const services = config.get('runtime.app.env.DX_DEBUG') ? fromHost(config) : fromIFrame(config);
```

Comment:
is there anything else that uses `DX_DEBUG`? Might want to be able to control that independently of monolithic mode, i've had instances where I've had to independently troubleshoot issues on both.

Reply:
This is just halo-app's own version of this, the main place it's used inside the sdk is here https://github.com/dxos/dxos/pull/4769/files#diff-e534edd3557d1d77a580677d740cee049c7b070804415b94be6e5d27e0cc4968R62.

Control what independent of monolithic mode?

Reply:
yeah that's all it does currently... it could definitely be more precise, though I was attempting to make it concise so as to make it easy to switch on whenever